File: lvl.Oidc.AuthorizationServer/Models/IdentityResource.cs
using FluentNHibernate.Automapping;
using FluentNHibernate.Automapping.Alterations;
using IdentityServer4.Models;
using lvl.Ontology;
using System.ComponentModel;

namespace lvl.Oidc.AuthorizationServer.Models
{
    [Description(nameof(IdentityResource))]
    public class IdentityResourceEntity : IdentityResource, IEntity
    {
        public int Id { get; set; }
    }

    public class IdentityResourceOverride : IAutoMappingOverride<IdentityResource>
    {
        public void Override(AutoMapping<IdentityResource> mapping)
        {
            mapping.Id(identity => identity.Name);
            mapping.HasMany(identity => identity.UserClaims).Element("Value");
        }
    }
}

File: @lvl.oidc/src/models/identity-resource.ts
import { IEntity } from '@lvl/core';

export class IdentityResource implements IEntity {
    public id: number;
    public enabled: boolean;
    public name: string;
    public displayName: string;
    public description: string;
    public required: boolean;
    public emphasize: boolean;
    public showInDiscoveryDocument: boolean;
    public userClaims: string[];
}
